test/unit/client.js

```diff
@@ -104,6 +115,44 @@ describe('Client userMuteStatus', function () {
 	});
 });
 
+describe('Client connectUser', () => {
+	let client;
+	beforeEach(() => {
+		client = new StreamChat('', { allowServerSideConnect: true });
+		client._setupConnection = () => Promise.resolve('_setupConnection');
+		client._setToken = () => Promise.resolve('_setToken');
+	});
+
+	it('should throw err for missing user id', async () => {
+		expect(() => client.connectUser({ user: 'user' }, 'token')).to.throw(
+			/The "id" field on the user is missing/,
+		);
+	});
+
+	it('should return a promise when called', async () => {
+		const promise = client.connectUser({ id: 'user' }, 'token');
+		expect(promise).to.be.a('promise');
+
+		const resolved = await promise;
+		expect(resolved).to.equal('_setupConnection');
+	});
+
+	it('should throw error if connectUser called twice on the client with different user', async () => {
```

Comment:
how are we handling changing the user ? new client in this case ?

Reply:
`client.disconnect()` should be called first

Reply:
new test is good, thanks